### PR TITLE
Upgrade buf yaml config and CI

### PR DESCRIPTION
**Pull request by emcfarlane**
This PR upgrades the config for buf modules, fixing lint checks, and updating the CI to push the main module. The config file is upgraded to v2 (follows this [guide](https://buf.build/docs/migration-guides/migrate-v2-config-files)). Ran `buf lint` and corrected the CEL expression on the `Interval` type. Uses [buf-action](https://github.com/bufbuild/buf-action) config to run build, checks (lint, format, breaking), and push on each feature branch and main to the BSR. Please see commit-by-commit for a breakdown of the steps.

There are a couple of behaviour changes in this PR. This implements the [default behaviour](https://github.com/bufbuild/buf-action?tab=readme-ov-file#default-behavior) as described in the buf-action repo. Every push will now sync with the BSR. On pull requests checks will be run. On merging to main the deleted branch will be archived and the latest on main synced.

### CODE REVIEW

**Review by rofrankel — APPROVED** [review]
Thanks for the fixes!

One nit:  it looks like this PR also makes some unrelated formatting changes.  I think the protos were previously clang-formatted with Google style.  Think we can preserve that, so that this PR only has contentful changes?

File: proto/aep-type/aep/type/phone_number.proto [hunk 1]
```diff
@@ -117,5 +115,5 @@ message PhoneNumber {
   // field is normally only set in conjunction with an E.164 number. It is held
   // separately from the E.164 number to allow for short code extensions in the
   // future.
-  string extension = 3 [(buf.validate.field).string = { max_len: 40 }];
-}
\ No newline at end of file
+  string extension = 3 [(buf.validate.field).string = {max_len: 40}];
```
Comment (rofrankel) [inline proto/aep-type/aep/type/phone_number.proto:118]: One nit: it looks like this PR also makes some unrelated formatting changes. I think the protos were previously clang-formatted with Google style. Think we can preserve that, so that this PR only has contentful changes?

(making this an actual comment on the PR so that it has to be resolved before merging)

**Review by rofrankel — APPROVED** [review]

File: proto/aep-type/aep/type/phone_number.proto (hunk 1 above)
Comment (emcfarlane) [reply proto/aep-type/aep/type/phone_number.proto:118]: Will revert the formatting changes, thanks.

**Review by rambleraptor — APPROVED** [review]
